File: Algorithms/1137.n-th-tribonacci-number/n-th-tribonacci-number.go
func tribonacci(n int) int {
    if n < 3 {
        if n == 0 {
            return 0
        } else {
            return 1
        }
    }

    tmp,x,y,z := 0,0,1,1
    for i := 3;i <= n;i++ {
        tmp = x + y + z
        x = y
        y = z
        z = tmp
    }

    return z
}

File: Algorithms/1137.n-th-tribonacci-number/n-th-tribonacci-number.java
class Solution {
    public int tribonacci(int n) {
        if (n < 3) return n == 0 ? 0 : 1;

    int tmp, x = 0, y = 1, z = 1;
    for (int i = 3; i <= n; ++i) {
      tmp = x + y + z;
      x = y;
      y = z;
      z = tmp;
    }
    return z;
    }
}